test/suites/integration/ram_share_test.go

```diff
@@ -0,0 +1,353 @@
+		}
+
+		deleteTaggedResources := func(
+			getResourcesResult *resourcegroupstaggingapi.GetResourcesOutput,
+			vpcLatticeClient *vpclattice.VPCLattice,
+			ramClient *ram.RAM,
+		) {
+			for _, mapping := range getResourcesResult.ResourceTagMappingList {
+				arn, err := arn2.Parse(*mapping.ResourceARN)
+				Expect(err).ToNot(HaveOccurred())
+
+				resourceType := strings.Split(arn.Resource, "/")[0]
+				switch resourceType {
+				case "servicenetwork":
+					deleteSnInput := &vpclattice.DeleteServiceNetworkInput{
+						ServiceNetworkIdentifier: aws.String(arn.String()),
+					}
+					_, err := vpcLatticeClient.DeleteServiceNetworkWithContext(ctx, deleteSnInput)
+					Expect(err).ToNot(HaveOccurred())
+				case "resource-share":
+					deleteShareInput := &ram.DeleteResourceShareInput{
+						ResourceShareArn: aws.String(arn.String()),
+					}
+					_, err := ramClient.DeleteResourceShareWithContext(ctx, deleteShareInput)
+					Expect(err).ToNot(HaveOccurred())
+				default:
+					Fail(fmt.Sprintf("Unknown resource type %s", resourceType))
+				}
+			}
+		}
+
+		// Delete secondary account AWS resources
+		secondaryGetResourcesResult, err := secondaryResourceGroupsClient.GetResources(getResourcesInput)
+		Expect(err).ToNot(HaveOccurred())
+		deleteTaggedResources(secondaryGetResourcesResult, secondaryVpcLatticeClient, secondaryRamClient)
+
+		// Delete primary account AWS resources
+		primaryGetResourcesResult, err := primaryResourceGroupsClient.GetResources(getResourcesInput)
+		Expect(err).ToNot(HaveOccurred())
+		deleteTaggedResources(primaryGetResourcesResult, primaryVpcLatticeClient, primaryRamClient)
+	})
+})
+
+func createSharedServiceNetwork(
+	serviceNetworkName string,
+	secondaryAccountId string,
+	primaryVpcLatticeClient *vpclattice.VPCLattice,
+	secondaryVpcLatticeClient *vpclattice.VPCLattice,
+	primaryRamClient *ram.RAM,
+	secondaryRamClient *ram.RAM,
+) *vpclattice.GetServiceNetworkOutput {
+	// Create secondary account's service network using randomName
+	createSNInput := &vpclattice.CreateServiceNetworkInput{
+		Name: aws.String(serviceNetworkName),
+		Tags: k8sTestTags,
+	}
+	createSNResult, err := secondaryVpcLatticeClient.CreateServiceNetworkWithContext(ctx, createSNInput)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Share service network to primary account using randomName
+	createShareInput := &ram.CreateResourceShareInput{
+		Name:                    aws.String(serviceNetworkName),
+		ResourceArns:            []*string{createSNResult.Arn},
+		Principals:              []*string{aws.String(config.AccountID)},
+		AllowExternalPrincipals: aws.Bool(true),
+		Tags:                    k8sRamTestTags,
+	}
+	_, err = secondaryRamClient.CreateResourceShareWithContext(ctx, createShareInput)
+	Expect(err).To(BeNil())
+
+	// Wait for resource share invitation to appear in primary account
+	var invitation *ram.ResourceShareInvitation = nil
+	Eventually(func(g Gomega) {
+		listInvitationsInput := &ram.GetResourceShareInvitationsInput{}
```

Comment:
Nit: I saw the vpc lattice canary python code did: `ram_client.get_resource_share_invitations(resourceShareArns=[resourceShareArn])`, Can you pass `resourceShareArns` in the `ram.GetResourceShareInvitationsInput`? and the response must have len(listInvitationsResult.ResourceShareInvitations)==1 or ==0 ?

Reply:
Thanks, will give this a shot